frontend/admin/src/js/components/home/HomePage.tsx

```diff
@@ -6,14 +6,22 @@ import {
 } from "@heroicons/react/24/outline";
 import CardLink from "@common/components/CardLink";
 import PageHeader from "@common/components/PageHeader";
-import { useLocation } from "@common/helpers/router";
+import { Navigate, useLocation } from "react-router-dom";
 import { getLocale } from "@common/helpers/localize";
 import { useApiRoutes } from "@common/hooks/useApiRoutes";
+import useAuth from "@common/hooks/useAuth";
+import { useAdminRoutes } from "../../adminRoutes";
 
 const HomePage: React.FC = () => {
   const intl = useIntl();
+  const adminRoutes = useAdminRoutes();
   const apiRoutes = useApiRoutes();
   const location = useLocation();
+  const { loggedIn } = useAuth();
+
+  if (loggedIn) {
+    return <Navigate to={adminRoutes.dashboard()} replace />;
```

Comment:
Mmmm, not an amazing reason. Just a easier escape out of the component since we are returning something. I could probably see about replacing with `useNavigate` for consistency

Reply:
<Navigate> is a thin wrapper around `useNavigate` (as it seems are most components in really clean libraries). But my sense is that the more that logic can be pulled out of the functional js part of a component and into components, the more we're doing things in the react way?

Reply:
Maybe that intuition needs to be put through the paces though, and maybe discarded :)

Reply:
Oh good find! Yes, they do recommend using `useNavigate` and only supply this component when you cannot use it!

> Having a component-based version of the useNavigate hook makes it easier to use this feature in a [React.Component](https://reactjs.org/docs/react-component.html) subclass where hooks are not able to be used.

REF: https://reactrouter.com/en/main/components/navigate

Reply:
Updated with: [prefer useNavigate over Navigate](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4597/commits/49a81f02f4304912a91c55a0b45f057c1aee0949)